src/components/Table.tsx

```diff
@@ -0,0 +1,592 @@
+import React, { useCallback, useEffect, useMemo, useState } from "react"
+import { useNavigate } from "react-router-dom"
+import { Link } from "react-router-dom"
+import { useVirtual } from "react-virtual"
+import {
+  QueryClient,
+  QueryClientProvider,
+  useInfiniteQuery,
+} from "@tanstack/react-query"
+import {
+  flexRender,
+  getCoreRowModel,
+  getSortedRowModel,
+  Row,
+  SortingState,
+  useReactTable,
+} from "@tanstack/react-table"
+import styled from "styled-components"
+
+import { useAppDispatch, useAppSelector } from "../app/hooks"
+import SortIconAsc from "../assets/SortIconAsc"
+import SortIconDesc from "../assets/SortIconDesc"
+import SortIconInactive from "../assets/SortIconInactive"
+import { setFilters } from "../features/search/searchSlice"
+
+const fetchSize = 25
+
+const queryClient = new QueryClient()
+
+const columns = [
+  {
+    accessorKey: "number",
+    header: "#",
+    size: 60,
+    cell: (info: any) => info.row.index + 1,
+  },
+  {
+    accessorKey: "accession",
+    header: "Entry",
+    cell: (info: any) => {
+      const value = info.getValue() as string
+
+      return (
+        <Link to={`/protein/${value}`} className="entry-link" target="_blank">
+          {value}
+        </Link>
+      )
+    },
+  },
+  {
+    accessorKey: "id",
+    header: "Entry Name",
+    cell: (info: any) => info.getValue(),
+  },
+  {
+    accessorKey: "gene",
+    header: "Genes",
+    cell: (info: any) => info.getValue(),
+  },
+  {
+    accessorKey: "organism_name",
+    header: "Organism",
+    cell: (info: any) => {
+      return (
+        <span className="organism-name">
+          {info.getValue() as React.ReactNode}
+        </span>
+      )
+    },
+  },
+  {
+    accessorKey: "subcellularLocation",
+    header: "Subcellular Location",
+    cell: (info: any) => {
+      const value = info.getValue() as string
+
+      if (value.length > 30) {
+        const words = value.split(" ")
+        const truncatedValue = words.slice(0, 2).join(" ")
+
+        return (
+          <span className="subcellular-location">
+            {truncatedValue}
+            {"..."}
+          </span>
+        )
+      }
+
+      return value
+    },
+  },
+  {
+    accessorKey: "length",
+    header: "Length",
+    cell: (info: any) => {
+      return info.getValue()
+    },
+  },
+]
+
+const Table = () => {
+  const searchQuery = useAppSelector((state) => state.search.searchQuery)
+
+  const selectedFilters = useAppSelector(
+    (state) => state.search.selectedFilters,
+  )
+
+  const navigate = useNavigate()
+  const dispatch = useAppDispatch()
+
+  const tableContainerRef = React.useRef<HTMLDivElement>(null)
+
+  const [sorting, setSorting] = useState<SortingState>([])
```

Comment:
why sorting is not in searchState?

Reply:
You have lost the main advantage or redux - ability to preserve downloaded search results